src/nix/derivation-instantiate.cc

```diff
@@ -0,0 +1,93 @@
+#include "command.hh"
+#include "common-args.hh"
+#include "store-api.hh"
+#include "derivations.hh"
+#include "local-fs-store.hh"
+#include "progress-bar.hh"
+
+#include <nlohmann/json.hpp>
+
+using namespace nix;
+
+static nlohmann::json storePathSetToJSON(const StorePathSet & paths, Store & store)
+{
+    auto res = nlohmann::json::object();
+    for (auto & path : paths) {
+        res[store.printStorePath(path)] = nlohmann::json::object();
+    }
+    return res;
+}
+
+// TODO deduplicate with other code also setting such out links.
```

Comment:
Would be nice to deduplicate this, but that can also be done in a future PR.

Reply:
See [commit ](https://github.com/NixOS/nix/pull/11657/commits/7f6d006bebfb56b2967165c7f2e8c940abafac24) which factors out `createOutLinks()`.